friends/src/App.js

```diff
@@ -0,0 +1,61 @@
+import React, { Component } from 'react';
+//import logo from './logo.svg';
+import { connect } from 'react-redux';
+import { fetchFriends } from './actions';
+import './App.css';
+
+class App extends Component {
+  state = {
+    username: '',
+    password: ''
+  }
+
+  handleChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    })
+  }
+
```

Comment:
you got a lot of code in this one, I know my saying is 'done is done' but if your able to import most of your logic like you did in the other assignment and keep this clean that's always a plus (don't stress over making it happen, this is just 'extra" feedback, for how to take this to the next level. You're doing fantastic overall